iati/tests/fixtures/versions.py

```diff
@@ -0,0 +1,428 @@
+  * `unknown` - pyIATI does not that this is an approved version of the IATI Standard that exists
+  * `possible` - a known or unknown version number
+  * `fullsupport` - pyIATI fully supports this version - this includes Schemas, Codelists, Rulesets, validation and more
+  * `partsupport` - pyIATI partially supports this version - this may mean that there is a Schema, but not the other components
+  * `single` - a fixture that will only parameterize with a single value
+  * `v0` - a version number with a major component of 0
+  * `v1` - a version number with a major component of 1
+  * `v2` - a version number with a major component of 2
+
+"""
+from decimal import Decimal
+import itertools
+import pytest
+import iati.version
+
+
+def generate_semver_list(major_components, minor_components, patch_components):
+    """Generate a list of SemVer-format values.
+
+    Params:
+        major_components (list of int): List of values to use as the Major Component in the generated Version Numbers.
+        minor_components (list of int): List of values to use as the Minor Component in the generated Version Numbers.
+        patch_components (list of int): List of values to use as the Patch Component in the generated Version Numbers.
+
+    Returns:
+        list of string: List of values in the SemVer format.
+
+    """
+    return [semver(version[0], version[1], version[2]) for version in itertools.product(major_components, minor_components, patch_components)]
+
+
+def iativer(integer, decimal):
+    """Construct an IATIver-format version number.
+
+    Args:
+        integer (int): The integer component of the version number.
+        decimal (int): The decimal component of the version number.
+
+    Returns:
+        str: An IATIver-format version number with the specified Integer and Decimal Components.
+
+    """
+    return str(integer) + '.0' + str(decimal)
+
+
+def semver(major_component, minor_component, patch_component):
+    """Construct an SemVer-format version number.
+
+    Args:
+        major_component (int): The major component of the version number.
+        minor_component (int): The minor component of the version number.
+        patch_component (int): The patch component of the version number.
+
+    Returns:
+        str: A SemVer-format version number with the specified Major, Minor and Patch Components.
+
+    """
+    return '.'.join([str(major_component), str(minor_component), str(patch_component)])
+
+
+def split_decimal(version_decimal):
+    """Split a Decimal version number into numeric representations of its components.
+
+    Args:
+        version_decimal (Decimal): A Decimal containing an IATI version number.
+
+    Returns:
+        list of int: A list containing numeric representations of the Integer and Decimal components.
+
+    """
+    integer_component = int(version_decimal)
+    decimal_component = int(version_decimal * 100) - 100
+
+    return [integer_component, decimal_component]
```

Comment:
It probably would be.

In terms of how it's used, there are tests such as `test_attribute_components_writable_valid_values()` that wish to modify a single component in the return value based on the value of another variable. For this use case, a `list` seems more appropriate, so this has been used as the return type for each "split a value" function.

It's possible that these utility functions may wish to move to the `version` module at some point. This might be a slightly better time to think about the mutability of the return types (since there will be more use cases to consider than utility functions for specific test fixtures). Thoughts?

Reply:
Happy to consider this at a later point.